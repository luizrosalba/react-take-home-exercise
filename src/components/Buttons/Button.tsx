export type ButtonProps  = {
    filter: string
    title: string
    setFilter: (filter: string)=> void
}
export default function Button({filter,title,  setFilter}: ButtonProps) {
    
  return (
    <button onClick={() => setFilter(filter)} className="text-gray-700">
    {title}
  </button>
  )
}
