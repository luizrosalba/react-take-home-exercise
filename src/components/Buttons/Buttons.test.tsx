import '@testing-library/jest-dom'
import {render, screen} from '@testing-library/react'
import Button from "./Button"

describe('Filter Test', () => {  

    it("Matches DOM Snapshot", () => {
        const filterMock = jest.fn()
        const domTree = render(<Button filter={'All'} title={'Placeholder'} setFilter={filterMock}/>);
        expect(domTree).toMatchSnapshot();
      });
  
      it("Clicking Button should call mock", () => {
        const filterMock = jest.fn()
        render(<Button filter={'All'} title={'Placeholder'} setFilter={filterMock}/>);
        const element = screen.queryByText('Placeholder')
        element?.click(); 
        expect(filterMock).toHaveBeenCalledTimes(1)
      });

  
  });