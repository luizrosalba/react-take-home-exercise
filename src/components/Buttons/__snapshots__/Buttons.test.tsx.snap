// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Test Matches DOM Snapshot 1`] = `
<div>
  <button
    class="px-4 rounded text-gray-700 dark:text-white"
  >
    Placeholder
  </button>
</div>
`;
